csrc/ops/alias.cpp

```diff
@@ -616,16 +652,52 @@ TensorView* cat(
           ", expected: ",
           ndims);
     }
-  }
 
-  if (cat_dim < 0) {
-    cat_dim += ndims;
+    bool cat_dim_empty = false;
+    // Check whether this input is possibly non-empty in any dimension
+    bool found_empty = false;
+    for (size_t dim : c10::irange(ndims)) {
+      if (extentIsEmpty(inp_dom[dim])) {
+        found_empty = true;
+        if (dim == cat_dim) {
+          cat_dim_empty = true;
+        }
+      }
+    }
+    all_inputs_empty = all_inputs_empty && found_empty;
+
+    if (cat_dim_empty) {
+      // Remove inputs that are empty in the cat dimension
+      continue;
+    }
+
+    inputs.push_back(inp);
+    inp_doms.emplace_back(inp_dom);
   }
 
-  NVF_CHECK(
-      cat_dim >= 0 && cat_dim < ndims, "Invalid dimension to cat: ", cat_dim);
+  if (inputs.empty()) {
+    // All tensors are empty in cat dimension
+    return set(orig_inputs.at(0));
+  } else if (all_inputs_empty) {
+    // All tensors are empty in at least one non-cat dimension. That means we
+    // can generate the output using full. The output size is computed using the
+    // size of the first original input except in the cat dim which is the sum
+    // of the extents of `inputs` in that dimension.
+    std::vector<Val*> shape(ndims, nullptr);
+    for (const std::vector<IterDomain*>& inp_dom : inp_doms) {
+      for (size_t dim : c10::irange(ndims)) {
+        Val* extent = inp_dom.at(dim)->getMaybeExpandedExtent();
+        if (dim == cat_dim) {
+          shape[dim] = SimplifyingIrBuilder::addExpr(shape[dim], extent);
+        } else if (shape[dim] == nullptr) {
+          shape[dim] = extent;
```

Comment:
I'm not sure what the correct behavior should be here. Can we just use the first extent for a non-cat dimension? Suppose the first cat input has size zero for that dimension, the output of the cat would also have zero for the dimension. Would that be expected?

Reply:
In that case it should be caught by the `inputs.empty()` condition at line 678, since all of the inputs should be empty. Actually though, I just realized that assumes that if one of the inputs has constant size 0 in one dimension that all the other inputs will have constant size zero in the same dimension. If some are symbolic this definition should be proof that they're equal by exact mapping, but they won't be detected at line 660. I guess what I should do instead is fire this condition if _any_ input has a zero non-cat dimension.

Reply:
> if one of the inputs has constant size 0 in one dimension that all the other inputs will have constant size zero in the same dimension.

Yeah, this was what I was thinking about. I'm not sure if that's actually allowed. Is it?

Reply:
We could have something like this I think:
```
tv0[ 2, 0 ]
tv1[ 3, i0 ]
tv2 = cat({tv0, tv1}, /*axis=*/0)
```
In this case we would normally exact map `tv0->axis(1)` with `tv1->axis(1)` so that `i0` must be 0. But now since we'll just translate this to `full({5, 0})` I don't think there's any such constraint, so maybe it would be legal to pass in a `tv1` with shape (3,2) and we would not hit an error.

Reply:
OK, makes sense.